### PR TITLE
Add Kotlin support

### PR DESCRIPTION
**Pull request by sdeleuze**
Could you please review this PR in order to allow us to be ready to merge this in master for the Kotlin 1.0 final release (1.0 beta 2 currently).

Closes gh-147

### CODE REVIEW

File: initializr/src/main/resources/templates/ApplicationTests.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package ${packageName}
+
+import org.junit.Test
+import org.junit.runner.RunWith
+${testImports}import org.springframework.boot.test.SpringApplicationConfiguration
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner
+
+@RunWith(SpringJUnit4ClassRunner::class)
+@SpringApplicationConfiguration(classes = arrayOf(${applicationName}::class))
+${testAnnotations}public class ${applicationName}Tests {
```
Comment (cy6erGn0m) [inline initializr/src/main/resources/templates/ApplicationTests.kt]: Isn't it already public by default?

File: initializr/src/main/resources/templates/starter-build.gradle [hunk 2]
```diff
@@ -1,6 +1,7 @@
 buildscript {
     ext {
         springBootVersion = '${bootVersion}'
+        kotlinVersion = '${kotlinVersion}'
```
Comment (cy6erGn0m) [inline initializr/src/main/resources/templates/starter-build.gradle]: Shouldn't it be conditional? `<% if (language == 'kotlin' .... `

File: initializr/src/test/groovy/io/spring/initializr/test/InitializrMetadataTestBuilder.groovy [hunk 3]
```diff
@@ -117,7 +117,7 @@ class InitializrMetadataTestBuilder {
 	}
 
 	InitializrMetadataTestBuilder addDefaultLanguages() {
-		addLanguage('java', true).addPackaging('groovy', false)
```
Comment (cy6erGn0m) [inline initializr/src/test/groovy/io/spring/initializr/test/InitializrMetadataTestBuilder.groovy]: `addPackaging` vs `addLanguage` ?

File: initializr/src/main/resources/templates/ApplicationTests.kt (hunk 1 above)
Comment (sdeleuze) [inline initializr/src/main/resources/templates/ApplicationTests.kt]: Indeed, still my Java habits ;-)

File: initializr/src/main/resources/templates/starter-build.gradle (hunk 2 above)
Comment (sdeleuze) [inline initializr/src/main/resources/templates/starter-build.gradle]: Good catch

File: initializr/src/test/groovy/io/spring/initializr/test/InitializrMetadataTestBuilder.groovy (hunk 3 above)
Comment (sdeleuze) [inline initializr/src/test/groovy/io/spring/initializr/test/InitializrMetadataTestBuilder.groovy]: I thought that was a mistake in current codebase, and replaced `addPackaging` by `addLanguage` for Groovy, could you please check this @snicoll ?

Comment (snicoll) [inline initializr/src/test/groovy/io/spring/initializr/test/InitializrMetadataTestBuilder.groovy]: indeed, I just fixed it. You can rebase your branch on master.

**Comment by sdeleuze** [comment]
@cy6erGn0m Thanks for your feedback, I have take them in account.
@snicoll I have rebased and pushed an updated version, including the appropriate `initializr-service/application.yml` configuration.

**Comment by sdeleuze** [comment]
I pushed a new version after @snicoll feedback, Kotlin version is now an env property in order to avoid modifying the meta model and being more consistent with Groovy support.

**Comment by cy6erGn0m** [comment]
Any news?

**Comment by snicoll** [comment]
I have reviewed the code and except a few minor modifications, it sounds good to me. We're now considering adding it to `start.spring.io` and this discussion can't only really happen when Kotlin final will be available.

**Pull request closed without merging**